contracts/AddressArrayLib.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Common : Address Array
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+
+library AddressArrayLib {
+
+    struct AddressArray {
+        address[] addresses;
+    }
+
+    //@dev Add element to array
+    function add(AddressArray storage array, address value) internal view returns (uint){
```

Comment:
Again something I would like to bring into discussion.
Should it be internal or private? Do we assume that we are going to inherit and override the function?

Reply:
Private methods are not accessible outside contracts. Methods in this library are consumed outside.